src/java.base/share/classes/jdk/internal/foreign/abi/SharedUtils.java

```diff
@@ -321,11 +321,17 @@ public static void checkNative(MemorySegment segment) {
         }
     }
 
+    @ForceInline
     public static long unboxSegment(MemorySegment segment) {
         checkNative(segment);
         return segment.address();
     }
 
+    @ForceInline
+    public static int unboxSegment32(MemorySegment segment) {
```

Comment:
Could you add a comment here:
```suggestion
    public static int unboxSegment32(MemorySegment segment) {
        // this cast to 'int' is safe, because we only call this method on 32-bit platforms, where we know the address of a segment is truncated to 32-bits.
```

Reply:
Done, pushed.